ffmpeg/ffmpeg.go

```diff
@@ -107,41 +108,162 @@ type TranscodeResults struct {
+	PixelFormatYUV422P16BE int = C.AV_PIX_FMT_YUV422P16BE
+	PixelFormatYUV444P16LE int = C.AV_PIX_FMT_YUV444P16LE
+	PixelFormatYUV444P16BE int = C.AV_PIX_FMT_YUV444P16BE
+	PixelFormatYUV420P12BE int = C.AV_PIX_FMT_YUV420P12BE
+	PixelFormatYUV420P12LE int = C.AV_PIX_FMT_YUV420P12LE
+	PixelFormatYUV422P12BE int = C.AV_PIX_FMT_YUV422P12BE
+	PixelFormatYUV422P12LE int = C.AV_PIX_FMT_YUV422P12LE
+	PixelFormatYUV444P12BE int = C.AV_PIX_FMT_YUV444P12BE
+	PixelFormatYUV444P12LE int = C.AV_PIX_FMT_YUV444P12LE
+)
+
+type ColorDepthBits int
+
+const (
+	ColorDepth8Bit  ColorDepthBits = 8
+	ColorDepth10Bit ColorDepthBits = 10
+	ColorDepth12Bit ColorDepthBits = 12
+)
+
+type ChromaSubsampling int
+
+const (
+	ChromaSubsampling420 ChromaSubsampling = iota
+	ChromaSubsampling422
+	ChromaSubsampling444
+)
+
+func (self PixelFormat) Properties() (ChromaSubsampling, ColorDepthBits, error) {
+	switch self.RawValue {
+	case C.AV_PIX_FMT_YUV420P:
+		return ChromaSubsampling420, 8, nil
+	case C.AV_PIX_FMT_YUYV422:
+		return ChromaSubsampling422, 8, nil
+	case C.AV_PIX_FMT_YUV422P:
+		return ChromaSubsampling422, 8, nil
+	case C.AV_PIX_FMT_YUV444P:
+		return ChromaSubsampling444, 8, nil
+	case C.AV_PIX_FMT_UYVY422:
+		return ChromaSubsampling422, 8, nil
+	case C.AV_PIX_FMT_NV12:
+		return ChromaSubsampling420, 8, nil
+	case C.AV_PIX_FMT_NV21:
+		return ChromaSubsampling420, 8, nil
+	case C.AV_PIX_FMT_YUV420P10BE:
+		return ChromaSubsampling420, 10, nil
+	case C.AV_PIX_FMT_YUV420P10LE:
+		return ChromaSubsampling420, 10, nil
+	case C.AV_PIX_FMT_YUV422P10BE:
+		return ChromaSubsampling422, 10, nil
+	case C.AV_PIX_FMT_YUV422P10LE:
+		return ChromaSubsampling422, 10, nil
+	case C.AV_PIX_FMT_YUV444P10BE:
+		return ChromaSubsampling444, 10, nil
+	case C.AV_PIX_FMT_YUV444P10LE:
+		return ChromaSubsampling444, 10, nil
+	case C.AV_PIX_FMT_YUV420P16LE:
+		return ChromaSubsampling420, 16, nil
+	case C.AV_PIX_FMT_YUV420P16BE:
+		return ChromaSubsampling420, 16, nil
+	case C.AV_PIX_FMT_YUV422P16LE:
+		return ChromaSubsampling422, 16, nil
+	case C.AV_PIX_FMT_YUV422P16BE:
+		return ChromaSubsampling422, 16, nil
+	case C.AV_PIX_FMT_YUV444P16LE:
+		return ChromaSubsampling444, 16, nil
+	case C.AV_PIX_FMT_YUV444P16BE:
+		return ChromaSubsampling444, 16, nil
+	case C.AV_PIX_FMT_YUV420P12BE:
+		return ChromaSubsampling420, 12, nil
+	case C.AV_PIX_FMT_YUV420P12LE:
+		return ChromaSubsampling420, 12, nil
+	case C.AV_PIX_FMT_YUV422P12BE:
+		return ChromaSubsampling422, 12, nil
+	case C.AV_PIX_FMT_YUV422P12LE:
+		return ChromaSubsampling422, 12, nil
+	case C.AV_PIX_FMT_YUV444P12BE:
+		return ChromaSubsampling444, 12, nil
+	case C.AV_PIX_FMT_YUV444P12LE:
+		return ChromaSubsampling444, 12, nil
+	default:
+		return 0, 0, ErrTranscoderPixelformat
+	}
+}
+
+type GetCodecStatus int
+
+const (
+	GetCodecInternalError  GetCodecStatus = -1
+	GetCodecOk             GetCodecStatus = 0
+	GetCodecNeedsBypass    GetCodecStatus = 1
+	GetCodecStreamsMissing GetCodecStatus = 2
+)
+
```

Comment:
Variable type GetCodecStatus seems like a function name.
How about this?

type CodecStatus int 

const (
	CodecStatusInternalError  CodecStatus = -1
	CodecStatusOk             CodecStatus = 0
	CodecStatusNeedsBypass    CodecStatus = 1
	CodecStatusMissing CodecStatus = 2
)